R/dsl.R

```diff
@@ -178,7 +175,9 @@ do_pkgdown <- function(...,
   get_stage("install") %>%
     add_step(step_install_deps(repos = repos))
 
-  if (isTRUE(build_only) || !deploy) {
+  needs_deploy <- !isTRUE(build_only) && ci_can_push()
+
+  if (isTRUE(needs_deploy)) {
```

Comment:
The `isTRUE()` looks superfluous, here and below.

Please double-check the logic.